libraries/AC_WPNav/AC_WPNav_OA.cpp

```diff
@@ -114,55 +140,56 @@ bool AC_WPNav_OA::update_wpnav()
         switch (oa_retstate) {
 
         case AP_OAPathPlanner::OA_NOT_REQUIRED:
+            // OA is no longer needed — restore original destination and optionally set next
             if (_oa_state != oa_retstate) {
                 // object avoidance has become inactive so reset target to original destination
-                if (!set_wp_destination_NEU_cm(_destination_oabak_neu_cm, _is_terrain_alt_oabak)) {
+                if (!set_wp_destination_NEU_m(_destination_oabak_neu_m, _is_terrain_alt_oabak)) {
                     // trigger terrain failsafe
                     return false;
                 }
 
                 // if path from destination to next_destination is clear
                 if (dest_to_next_dest_clear && (oa_ptr->get_options() & AP_OAPathPlanner::OA_OPTION_FAST_WAYPOINTS)) {
                     // set next destination if non-zero
-                    if (!_next_destination_oabak_neu_cm.is_zero()) {
-                        set_wp_destination_next_NEU_cm(_next_destination_oabak_neu_cm);
+                    if (!_next_destination_oabak_neu_m.is_zero()) {
+                        set_wp_destination_next_NEU_m(_next_destination_oabak_neu_m);
                     }
                 }
                 _oa_state = oa_retstate;
             }
 
-            // ensure we stop at next waypoint
+            // Prevent transitioning past this waypoint if path to next is unclear
             // Note that this check is run on every iteration even if the path planner is not active
             if (!dest_to_next_dest_clear) {
                 force_stop_at_next_wp();
             }
             break;
 
         case AP_OAPathPlanner::OA_PROCESSING:
+            // Allow continued movement while OA path is processing if fast-waypointing is enabled
             if (oa_ptr->get_options() & AP_OAPathPlanner::OA_OPTION_FAST_WAYPOINTS) {
                 // if fast waypoint option is set, proceed during processing
                 break;
             }
             FALLTHROUGH;
 
         case AP_OAPathPlanner::OA_ERROR:
-            // during processing or in case of error stop the vehicle
-            // by setting the oa_destination to a stopping point
+            // OA temporarily failing — stop vehicle at current position
             if ((_oa_state != AP_OAPathPlanner::OA_PROCESSING) && (_oa_state != AP_OAPathPlanner::OA_ERROR)) {
                 // calculate stopping point
-                Vector3f stopping_point;
-                get_wp_stopping_point_NEU_cm(stopping_point);
-                _oa_destination = Location(stopping_point, Location::AltFrame::ABOVE_ORIGIN);
+                Vector3f stopping_point_neu_m;
+                get_wp_stopping_point_NEU_m(stopping_point_neu_m);
+                _oa_destination = Location(stopping_point_neu_m, Location::AltFrame::ABOVE_ORIGIN);
```

Comment:
I think the Location object is still initialised using cm so we need an extra "*100" here.

_oa_destination = Location(stopping_point_neu_m * 100.0, Location::AltFrame::ABOVE_ORIGIN);